extensions/github-repository-search/src/index.tsx

```diff
@@ -11,55 +11,113 @@ import {
   const [debouncedSearchText] = useDebounce(searchText, 200);
-  const { data, error, isLoading: isLoadingRepositories } = useRepositories(debouncedSearchText);
+  const {
+    data: repositories,
+    error: searchRepositoriesError,
+    isLoading: isLoadingRepositories,
+  } = useRepositories(debouncedSearchText, searchFilter);
+  const { data: userData, error: fetchUserDataError } = useUserData();
+
   const {
     repositories: visitedRepositories,
     visitRepository,
     isLoading: isLoadingVisitedRepositories,
   } = useVisitedRepositories();
 
-  if (error) {
+  if (searchRepositoriesError) {
     showToast({
       style: Toast.Style.Failure,
       title: "Failed searching repositories",
-      message: error instanceof Error ? error.message : String(error),
+      message:
+        searchRepositoriesError instanceof Error ? searchRepositoriesError.message : String(searchRepositoriesError),
+    });
+  }
+  if (fetchUserDataError) {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "Failed fetching user and organizations",
+      message: fetchUserDataError instanceof Error ? fetchUserDataError.message : String(fetchUserDataError),
     });
   }
 
   const isLoading = searchText !== debouncedSearchText || isLoadingVisitedRepositories || isLoadingRepositories;
 
   return (
-    <List isLoading={isLoading} onSearchTextChange={setSearchText}>
+    <List
+      isLoading={isLoading}
+      onSearchTextChange={setSearchText}
+      searchBarAccessory={<FilterDropdown userData={userData} onFilterChange={setSearchFilter} />}
+    >
       <List.Section
         title="Visited Repositories"
         subtitle={visitedRepositories ? String(visitedRepositories.length) : undefined}
       >
         {visitedRepositories
-          ?.filter((r) => r.nameWithOwner.includes(searchText ?? ""))
+          ?.filter(
+            (r) =>
+              r.nameWithOwner.includes(searchText ?? "") &&
+              r.nameWithOwner.match(
+                // Converting query filter string to regexp:
+                `${searchFilter?.replaceAll(/org:|user:/g, "").replaceAll(" ", "|")}/.*`
+              )
+          )
           .map((repository) => (
             <RepositoryListItem key={repository.id} repository={repository} onVisit={visitRepository} />
           ))}
       </List.Section>
-      <List.Section title="Found Repositories" subtitle={data ? String(data.repositoryCount) : undefined}>
-        {data?.nodes?.map((repository) => (
+      <List.Section
+        title="Found Repositories"
+        subtitle={repositories ? String(repositories.repositoryCount) : undefined}
+      >
+        {repositories?.nodes?.map((repository) => (
           <RepositoryListItem key={repository.id} repository={repository} onVisit={visitRepository} />
         ))}
       </List.Section>
     </List>
   );
 }
 
+function FilterDropdown({
+  userData,
+  onFilterChange,
+}: {
+  userData: UserDataResponse["viewer"] | undefined;
```

Comment:
Isn't that the `viewer`?
```suggestion
  viewer?: UserDataResponse["viewer"];
```

Reply:
Sure it is.
I renamed it because I didn't think 'viewer' was a great name anymore when you detach it from the context of the request, but I guess you still have it in the type information